cabot_alert_email/models.py

```diff
@@ -1,48 +1,77 @@
 from os import environ as env
 
 from django.conf import settings
-from django.core.mail import send_mail
-from django.core.urlresolvers import reverse
+from django.core.mail import EmailMessage
 from django.template import Context, Template
 
 from cabot.cabotapp.alert import AlertPlugin
 
-import requests
-import logging
 
 email_template = """Service {{ service.name }} {{ scheme }}://{{ host }}{% url 'service' pk=service.id %} {% if service.overall_status != service.PASSING_STATUS %}alerting with status: {{ service.overall_status }}{% else %}is back to normal{% endif %}.
 {% if service.overall_status != service.PASSING_STATUS %}
 CHECKS FAILING:{% for check in service.all_failing_checks %}
   FAILING - {{ check.name }} - Type: {{ check.check_category }} - Importance: {{ check.get_importance_display }}{% endfor %}
+{% if links %}Related links for the failing checks:\n{{ links|safe }}.{% endif %}
 {% if service.all_passing_checks %}
 Passing checks:{% for check in service.all_passing_checks %}
   PASSING - {{ check.name }} - Type: {{ check.check_category }} - Importance: {{ check.get_importance_display }}{% endfor %}
 {% endif %}
 {% endif %}
 """
 
+
 class EmailAlert(AlertPlugin):
     name = "Email"
     author = "Jonathan Balls"
 
     def send_alert(self, service, users, duty_officers):
-        emails = [u.email for u in users + duty_officers if u.email]
+        """
+        Send an email to the specified users with the service status and (possibly) Grafana panel images/links.
+        """
+        emails = [u.email for u in users if u.email] + [u.email for u in duty_officers if u.email]
         if not emails:
             return
-        c = Context({
-            'service': service,
-            'host': settings.WWW_HTTP_HOST,
-            'scheme': settings.WWW_SCHEME
-        })
+
+        links = []
+        images = {}
         if service.overall_status != service.PASSING_STATUS:
             subject = '%s status for service: %s' % (
                 service.overall_status, service.name)
+
+            failing_metrics_checks = service.status_checks \
```

Comment:
This is used twice. Once in email plugin and once in hipchat plugin. We should make this a top-level API for service.
```python
service.get_failing_checks()
```